src/sccz80/codegen.c

```diff
@@ -1114,14 +1114,19 @@ void gen_call(int arg_count, const char *name, SYMBOL *sym)
     ot("call\t"); outname(name, dopref(sym)); nl();
 }
 
-void gen_shortcall(Type *functype, int rst, int value) 
+void gen_shortcall(Type *functype, int rst, int value, char hl)
 {
-    if (functype->return_type->kind == KIND_LONGLONG) {
-        ol("ld\tbc,__i64_acc");
-        push("bc");
+    if (hl) {
+        outfmt("\tld hl,\t%d\n", value);
```

Comment:
Call `vconst(val)` - it will might up an opportunity for optimisation by the peep hole rules.